frontend/test/pytest/test_template.py

```diff
@@ -690,8 +1187,326 @@ def qnn(weights, inputs):
+    jitted_fn = qjit(interpreted_fn)
+
+    interpreted_result = interpreted_fn()
+    jitted_result = jitted_fn()
+
+    assert np.allclose(interpreted_result, jitted_result)
+
+
+def test_phase_adder(backend):
+    """Test PhaseAdder."""
+    x = 8
+    k = 5
+    mod = 15
+
+    x_wires = [0, 1, 2, 3]
+    work_wire = [4]
+
+    def phase_adder():
+        qml.BasisEmbedding(x, wires=x_wires)
+        qml.QFT(wires=x_wires)
+        qml.PhaseAdder(k, x_wires, mod, work_wire)
+        qml.adjoint(qml.QFT)(wires=x_wires)
+        return qml.sample(wires=x_wires)
+
+    device = qml.device(backend, wires=range(5), shots=2)
+    interpreted_fn = qml.QNode(phase_adder, device)
+    jitted_fn = qjit(interpreted_fn)
+
+    interpreted_result = interpreted_fn()
+    jitted_result = jitted_fn()
+
+    assert np.allclose(interpreted_result, jitted_result)
+
+
+@pytest.mark.xfail(reason="Qutrit operators not supported on lightning.")
+def test_qutrit_basis_state_preparation(backend):
+    """Test QutritBasisStatePreparation."""
+    basis_state = [0, 1]
+    wires = [0, 1]
+    obs = np.array([[1, 0, 0], [0, 2, 0], [0, 0, 3]])
+
+    def qutrit_basis_state_preparation(state, obs):
+        qml.QutritBasisStatePreparation(state, wires)
+
+        return [qml.expval(qml.THermitian(A=obs, wires=i)) for i in range(3)]
+
+    device = qml.device(backend, wires=2)
+    interpreted_fn = qml.QNode(qutrit_basis_state_preparation, device)
+    jitted_fn = qjit(interpreted_fn)
+
+    interpreted_result = interpreted_fn(basis_state, obs)
+    jitted_result = jitted_fn(basis_state, obs)
+
+    assert np.allclose(interpreted_result, jitted_result)
+
+
+def test_cosine_window(backend):
+    """Test cosine window."""
+
+    def cosine_window():
+        qml.CosineWindow(wires=[0, 1])
+        return qml.probs(wires=[0, 1])
+
+    device = qml.device(backend, wires=2)
+    interpreted_fn = qml.QNode(cosine_window, device)
+    jitted_fn = qjit(interpreted_fn)
+
+    interpreted_result = interpreted_fn()
+    jitted_result = jitted_fn()
+
+    assert np.allclose(interpreted_result, jitted_result)
+
+def test_basis_rotation(backend):
+    """Test BasisRotation"""
+
+    unitary_matrix = jnp.array(
+        [
+            [0.51378719 + 0.0j, 0.0546265 + 0.79145487j, -0.2051466 + 0.2540723j],
+            [0.62651582 + 0.0j, -0.00828925 - 0.60570321j, -0.36704948 + 0.32528067j],
+            [-0.58608928 + 0.0j, 0.03902657 + 0.04633548j, -0.57220635 + 0.57044649j],
+        ]
+    )
+
+    def basis_rotation(unitary_matrix, check):
+        qml.BasisState(qml.math.array([1, 1, 0]), wires=[0, 1, 2])
+        qml.BasisRotation(
+            wires=range(3),
+            unitary_matrix=unitary_matrix,
+            check=check,
+        )
+        return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+    device = qml.device(backend, wires=3)
+    interpreted_fn = qml.QNode(basis_rotation, device)
+    jitted_fn = qjit(interpreted_fn, static_argnums=1)
```

Comment:
I would have expected to be static_argnums = 0
Is this correct?

Reply:
`static_argnums = 0` is correct. The `check` variable is the one that needs to be static.

Reply:
`static_argnums = 1` is correct. We want the `check` variable to be static.